### DISCUSSION TITLE
tokens.json file - how to?

### DISCUSSION THREAD

**Discussion by lmormol** [terrazzoapp/terrazzo#121] [General]
Hi 👋

[examples/adobe/tokens.json](https://github.com/drwpow/cobalt-ui/blob/main/examples/adobe/tokens.json)

I'm wondering how did you generate such a beautiful JSON file like this ? 🤔

Is there a possibility to see the Figma file with this variables setup ?
Did you use any current existing Figma plugins to create that tokens.json ?

BR, Ł

**Comment by drwpow** [comment]
No tools; just wrote a Node.js script to pull values from Adobe’s open-source Spectrum code on GitHub and put it in a JSON file. Then went back in and fixed it up manually in VS Code. A lot of ad-hoc string mashing and RegEx 😄

There’s no Figma file, though I am actively working on better Figma tooling to make that export process to design tokens JSON easier.

**Reply by lmormol** [reply]
Thanks for an answer. I really appreciate your work on Cobalt UI!

One short note for a future readers ...

Few months later I've found Figma plugin called [TokensBrücke](https://www.figma.com/community/plugin/1254538877056388290/tokensbrucke) which started support COBALT UI and allows for an export handling collection modes.

**Reply by drwpow** [reply]
TokensBrucke is amazing! It’s actually listed in the docs now as a recommended Figma integration (Cobalt probably won’t ever have a Figma plugin, or at least not for a very long time)

**Reply by drwpow** [reply]
Small update—a lot of the public design systems I’ve moved here—https://github.com/terrazzoapp/dtcg-examples. There’s also a `dtcg-examples` package on npm for easier consumption.
